### ISSUE TITLE
[Bug]: Setting house property 16 to > 0 prevents callback 1B from running at the start of the game.

### ISSUE THREAD

**Issue by PikkaBird** [OpenTTD/OpenTTD#12986]
### Version of OpenTTD

20241008

### Expected result

As is the norm these days, I am creating houses that orient to roads, and storing that orientation in the animation frame to avoid houses "rebuilding" if roads are removed or added by the player.

This is all well and good, but I am now wanting to add elements that update with the periodic processing. OpenTTD provides a property - 0x16 - to only rerandomise every n-th processing cycle, for animations etc that should update less frequently than the default. Callback 1B, which sets the animation frame, is only called during the processing cycle, and only updates every n-th cycle when property 16 is set to > 0.

However, 1B also runs, simultaneously, for all houses at the game start - which is important for intialising houses correctly.

### Actual result

Setting house property 16 to > 0 prevents callback 1B from running at the start of the game - that is, it appears to skip cycles, then run, rather than running a cycle, then skipping.
![Unnamed, 1st Jan 1950](https://github.com/user-attachments/assets/6bec92af-7198-423c-93fd-9f9c5cfbc8d6)

This church uses cb 1B to set its sprites, with paths to neigbouring roads. It also has little transport-tycoon-people sprites which rerandomise from time to time. But using property 16 to slow down the rerandomisation causes the animation frame callback not to run on initialisation, resulting in the sprites being broken as seen until the first time the periodic processing runs.

### Steps to reproduce

A demonstration GRF can be provided if required.

**Comment by PeterN** [comment]
Even if the processing delay is not applied to the first cycle, the house can be visible before the tile's tile loop processing is handled.

So while it might be possible to ignore the processing delay for the first cycle, I think the graphics layout should still return a valid state before animation control occurs.

**Comment by PikkaBird** [comment]
Certainly, I can (and will) pretty up the code so that it shows real, if not necessarily correct, sprites when uninitialized. Houses are usually briefly visible before being processed when manually placed in the scenario editor, which is acceptable, and when constructed during gameplay, which can be handled with a construction stage sprite.

But when generating a new map, it seems that all houses are usually processed, and cb 1B run, once before the game start, and that's what setting prop 16 to > 0 breaks.

**Comment by PikkaBird** [comment]
Here's perhaps a better illustration, with the game paused at game start. The yellow and orange houses all have code that detects the road, stores the result in the animation frame with cb 1B, then picks graphics accordingly. The yellow houses have prop 16 set to 0, and show the correct graphics immediately. The orange houses have prop 16 set to >0, and show the uninitialized default state until the first time the tile loop runs (which can be quite a while, depending on the setting of prop 16).

I could, in theory, live-detect the road until the tile is initialized, as a workaround for the bug. But it still feels like it *is* a bug.

![Unnamed, 1st Jan 1920#1](https://github.com/user-attachments/assets/12132835-befc-403f-9410-a5bba33ece30)

**Comment by PeterN** [comment]
Hmm, am I imagining seeing a GRF for this...

**Comment by 2TallTyler** [comment]
> A demonstration GRF can be provided if required.

Maybe @PikkaBird can provide one. 🙂

**Comment by PikkaBird** [comment]
[sunshine_houses.zip](https://github.com/user-attachments/files/17671553/sunshine_houses.zip)

Here's my WIP GRF. The south tile of the church and the houses with the orange ground tiles in transparent mode are the ones affected. They currently run the callback and update to the correct frame after ~2 in game weeks.

**Comment by PikkaBird** [comment]
An uninitialized period for houses built during the game is more tolerable IMO, since they can be given a variation-agnostic initial construction stage.

**Comment by PeterN** [comment]
Hmm, most other features with animation also have an animation trigger when they are built. But houses do not.

Rather than bodging the tileloop trigger to be called early, perhaps there should just be a built trigger instead.

I don't know the reasoning why there isn't one. Triggers seem to generally be a bit different for houses, than the other features.

**Comment by zephyris** [comment]
Given the new ability to place houses, I've been facing this issue with my "Full English" set... Buildings rotate slowly after they're placed. Same following town expansion in the scenario editor. I think, but haven't thoroughly checked, that it's also an issue with normal construction - my workaround was non-orientation sensitive construction sprites.

An animation check trigger on built would fix this. Though, should probably check if it affects a town set which uses animation frames for animation rather than persistent storage.
